corehq/apps/locations/fixtures.py

```diff
@@ -191,24 +194,31 @@ def _get_fixture_node(self, fixture_id, restore_user, locations_queryset,
         return root_node
 
 
-def should_sync_hierarchical_fixture(project):
-    # Sync hierarchical fixture for domains with fixture toggle enabled for migration and
-    # configuration set to use hierarchical fixture
-    # Even if both fixtures are set up, this one takes priority for domains with toggle enabled
-    return (
-        project.uses_locations and
-        toggles.HIERARCHICAL_LOCATION_FIXTURE.enabled(project.name) and
-        LocationFixtureConfiguration.for_domain(project.name).sync_hierarchical_fixture
-    )
+def should_sync_hierarchical_fixture(project, app):
+    if (not project.uses_locations
+            or not toggles.HIERARCHICAL_LOCATION_FIXTURE.enabled(project.name)):
+        return False
 
+    if app and app.location_fixture_restore in SYNC_HIERARCHICAL_FIXTURE:
+        return True
 
-def should_sync_flat_fixture(project):
-    # Sync flat fixture for domains with conf for flat fixture enabled
-    # This does not check for toggle for migration to allow domains those domains to migrate to flat fixture
-    return (
-        project.uses_locations and
-        LocationFixtureConfiguration.for_domain(project.name).sync_flat_fixture
-    )
+    if app and app.location_fixture_restore != DEFAULT_LOCATION_FIXTURE_OPTION:
+        return False
+
+    return LocationFixtureConfiguration.for_domain(project.name).sync_hierarchical_fixture
+
+
+def should_sync_flat_fixture(project, app):
+    if not project.uses_locations:
+        return False
+
+    if app and app.location_fixture_restore in SYNC_FLAT_FIXTURES:
+        return True
+
+    if app and app.location_fixture_restore != DEFAULT_LOCATION_FIXTURE_OPTION:
+        return False
+
+    return LocationFixtureConfiguration.for_domain(project.name).sync_flat_fixture
```

Comment:
`restore_state.params.app` is an app build, right?  The nice thing about that is you could tie fixture dependency to app versions, allowing projects like ICDS to roll out the flat fixture to users without a transition period of sending down both (though I imagine we'd still want that capability for the app managers during conversion).

You might need to do a migration to update old app versions to whatever the current domain level setting is.  Have you thought about consolidating the two means of specifying fixture format?

Reply:
> The nice thing about that is you could tie fixture dependency to app versions, allowing projects like ICDS to roll out the flat fixture to users without a transition period of sending down both

This is exactly what this PR enables. or at least intends to. maybe we should offline if that isn't clear?

Reply:
No, that makes sense - from the description I had assumed that it was focused only on per-app settings (ie, for mutiple apps on the domain), rather than per app version.

Is this a fair summary of the intended process?
* All apps on the domain use hierarchical, and the domain setting is hierarchical only
* App builders turn on both fixtures for testing during the conversion process, but users continue to get only hierarchical, because they're on an old version
* App builders release a new version, which syncs only the flat fixture.
* Once all apps have been converted in this manner, and all users are on new enough versions of the app, the domain setting is switched to flat.

Reply:
> No, that makes sense - from the description I had assumed that it was focused only on per-app settings (ie, for mutiple apps on the domain), rather than per app version.

My understanding is that when a build is made, the app doc is essentially copied into a build doc. that build doc would then contain this flag. When "app aware" restores happens it's actually a "build aware" restore, so it would grab that build's settings.

> Is this a fair summary of the intended process?

Close. I would like to get rid of LocationFixtureConfiguration as well, (which means updating all builds for all domains that have this flag). If that can be done, then I'd like to get rid of that page/option totally. I'm not doing that in this PR because that migration is a little tricky. I imagine it will look  like:

1) go back and update all builds/apps in all domains with the hierarchical flag on
2) figure out if all of those users are using a commcare version featuring app aware restores
3a) If no, we will need to wait until they are on a version that does that, or always keep that domain syncing hierarchical version
3b) if yes, we can remove thier locationfixtureconfiguration and only use the app setting.

Reply:
Cool, that sounds awesome!